robot/impl/local_robot_test.go

```diff
@@ -3523,12 +3523,27 @@ func TestRestartModule(t *testing.T) {
 	logger := logging.NewTestLogger(t)
 	simplePath := rtestutils.BuildTempModule(t, "examples/customresources/demos/simplemodule")
 	mod := &config.Module{Name: "restartSingleModule-test", ExePath: simplePath, Type: config.ModuleTypeLocal}
-	r := setupLocalRobot(t, ctx, &config.Config{Modules: []config.Module{*mod}}, logger)
-	test.That(t, mod.LocalVersion, test.ShouldBeEmpty)
 
-	// test restart. note: we're not testing that the PID rolls over because we don't have access to
-	// that state. 'no error' + 'version incremented' is a cheap proxy for that.
-	err := r.(*localRobot).restartSingleModule(ctx, mod)
-	test.That(t, err, test.ShouldBeNil)
-	test.That(t, mod.LocalVersion, test.ShouldResemble, "0.0.1")
+	t.Run("isRunning=false", func(t *testing.T) {
+		// Config is missing the module to exercise the case where the module is not currently
+		// running and needs to be added instead of modified. Ideally we'd do an e2e test
+		// of the RestartModule call, but I think we don't have enough control of the process
+		// manager to do that simply.
```

Comment:
To confirm, we cannot simply create a scenario where the module crashes immediately?

Reply:
I could clone `examples/customresources/demos/simplemodule` + make it crash, but I'm not sure the existing (localRobot -> moduleManager -> processManager) would bubble that up in a way I can detect

another way to do this would be to write auditable dummies for module + process manager -- didn't want to build that much infra without talking to you, and took a while to get it all in my head, but RestartModule has failed often enough that I wouldn't mind doing that work now

Reply:
test module has [a build-in command that kills the underlying process](https://github.com/viamrobotics/rdk/blob/8241c77e989cfea5a47e1eddb191d1e8684fc560/module/testmodule/main.go#L130-L134)

we use this in various tests that need to crash right away - [here's one example](https://github.com/viamrobotics/rdk/blob/8241c77e989cfea5a47e1eddb191d1e8684fc560/module/modmanager/manager_test.go#L904-L990) - is this functionality sufficient for a test that calls `RestartModule` directly?

Reply:
ah we capture logs -- okay will copy this pattern + then retag you

Reply:
ObservedTestLogger didn't capture the 'exited too quickly' line, guessing because I'm passing the logger to localRobot and mod manager uses a different one? did something slightly different:
- my test module is just a bash script that exits right away. this is the case we want to check anyway, 'user made a mistake in module and wants to restart'
- the bash script writes a file when it runs which we can check in the test